src/QMCWaveFunctions/BsplineFactory/SplineR2R.h

```diff
@@ -79,6 +88,37 @@ class SplineR2R : public BsplineSet
     KeyWord    = "SplineR2R";
   }
 
+  void setOrbitalSetSize(int norbs) override;
+
+  void checkObject() const override;
+
+  
+  // Gives the number of splined SPOs
+  int getOrbitalSetSize() const override
+  {
+    //const int nsplines = ( SplineInst == nullptr ) ? 0 : SplineInst->num_splines();
+    return m_ntot_orbs;       
+  }
+  
+  
+  // @TODO (JPT) 28.09.2021 Cross check w/ LCAO
+  // Gives the number of splined SPOs
+  int getBasisSetSize() const override
+  {
+    const int nsplines = ( SplineInst == nullptr ) ? 0 : SplineInst->num_splines();
+    //std::cerr << "  Nsplines = " << nsplines << "\n";
+    return ( SplineInst == nullptr ) ? 0 : SplineInst->num_splines();   
+  }
+
+  // Accumulate the total rotation
+  void storeParamsBeforeRotation() override
+  {
+    std::cerr << "Function SplineR2R::storeParamsBeforeRotation() not implemented!\n";
+  }
+  
+  void applyRotation(const ValueMatrix_t& rot_mat, bool use_stored_copy) override;
```

Comment:
Please make a PR with only applyRotation being added and a unit test.
We are missing proper documentation for this function ins `SPOSet.h`.
```
rot_mat
```
should be of size num_occupied x OrbitalSetSize.